### PR TITLE
toolchain: bump to r27c

### PR DESCRIPTION
**Pull request by licy183**

### CODE REVIEW

**Comment by fornwall** [comment]
If you want to you can drop the gdal patch (https://github.com/termux/termux-packages/blob/master/packages/gdal/gdal.cmake.patch) which works around a crash that is fixed in r27c. But that's just a cleanup, can also be done as a separate follow up PR.

**Review by truboxl — APPROVED** [review]

**Comment by fornwall** [comment]
@licy183 Is this ready to merge, or any known (or potential) issue to look into?

**Comment by finagolfin** [comment]
Hold on, I have some Swift-specific changes in ndk-sysroot that I will add in a commit, then you can merge.

File: ndk-patches/27c/c++-v1-module.modulemap.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+--- a/usr/include/c++/v1/module.modulemap	2024-11-08 20:03:22
++++ b/usr/include/c++/v1/module.modulemap	2024-11-08 21:27:52
+@@ -1537,6 +1537,7 @@
+ module std_private_memory_uninitialized_algorithms        [system] {
+   header "__memory/uninitialized_algorithms.h"
+   export std_private_algorithm_copy
++  export std_private_memory_voidify
```
Comment (finagolfin) [inline ndk-patches/27c/c++-v1-module.modulemap.patch:7]: I've reported this upstream, swiftlang/swift#76325, probably a mistake that will be fixed in the next re-org mentioned there.

File: ndk-patches/27c/c++-v1-stdlib.h.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
+--- a/usr/include/c++/v1/stdlib.h	2024-11-08 20:01:18
++++ b/usr/include/c++/v1/stdlib.h	2024-11-08 20:03:38
+@@ -109,6 +109,7 @@
+ #    endif
+ 
+ // MSVCRT already has the correct prototype in <stdlib.h> if __cplusplus is defined
++#    if !defined(__swift__)
```
Comment (finagolfin) [inline ndk-patches/27c/c++-v1-stdlib.h.patch:7]: Since this is only defined for Swift, should have no effect on any other project.

**Comment by finagolfin** [comment]
Fixed a typo in my commit message, ready to go.

**Comment by licy183** [comment]
I'll merge this tomorrow afternoon (UTC+8) if there is no more review. Thanks!

**Pull request merged**